### ISSUE TITLE
Three Dimensional color gradient

### ISSUE THREAD

**Issue by leonbohmann** [helix-toolkit/helix-toolkit#1307]
Hello!

I was searching on the web for information about applying color to meshes with SharpDX Helixtoolkit.

Still, I cant figure out, how to apply a gradient brush onto a box I built. 
```c#

MeshBuilder builder = new MeshBuilder(true, true, true);
builder.AddBox(node.Deflected, 0.05f, 0.05f, 0.05f, BoxFaces.All);
var BoxGeometryModel = new MeshGeometryModel3D();
BoxGeometryModel.Geometry = builder.ToMeshGeometry3D();

```
I know a specific value of each corner and I also have the appropate Color for each corner. Ideally, what I want to do is to have a gradient inside the Box with the specified color on each corner of it while interpolating between them, so when I'd slice the cube, I would have a gradient on the sliced face as well. For now, I'd be satisfied if I could apply a gradient to each face individually though.

How do I achieve that? Do I really need to create a Texture and apply that to the Box? How would I go about that?

Thanks in advance!!

**Comment by leonbohmann** [comment]
Ok I already solved the first stage of the problem myself!
By assigning a `VertColorMaterial` to the GeometryModel and setting the Colors of the Mesh as follows:
```c#
Model.Colors = new Color4Collection(new[] { 36-times: Colors.Green.ToColor4() });
```
i get the cube, which is show in the picture below. The Material only is on the outside of the box.

What makes this a little harder is the fact, that I know the value of some points in between (marked red)! 

![grafik](https://user-images.githubusercontent.com/13386367/74105423-d7a24880-4b5d-11ea-8c41-31048c3d3022.png)

Now, by only assigning the color on the vertices of the mesh, I will not be able to specify the desired color for the points in between, am I? How do I go about that? Is there a way to build the mesh differently, so I get the desired vertices created at these points?

Thanks in advace, again! ;)

**Comment by holance** [comment]
You need to generate your own custom box mesh and add vertices on box face.

**Comment by leonbohmann** [comment]
Thanks!

Is there a simple way or do I have to generates the rectangles on each box face on my own?
Or can I just add a vertex manually? That wont work, because its not connected, right?

**Comment by holance** [comment]
If your model is fixed, you can manually generate the mesh yourself or use some modelling tool to create a mesh, then import into helixtoolkit.

**Comment by leonbohmann** [comment]
Hi again,

thanks for the reply!

If I am generating the Color4Collection on my own, I have to add the Colors which are used by the VertColorMaterial in the same order as the positions?

**Comment by leonbohmann** [comment]
Or should I use the PositionColorMaterial? How does that work? I tried applying that but then I dont get the correct color when applying that on the same Model..

**Comment by holance** [comment]
Yes, vertex color needs to be same order as vertex value in collection. PositionColorMaterial is only used for debugging purpose. I don't think it is what you need.

**Comment by leonbohmann** [comment]
Okay! Thanks for the reply!

One last thing for now:
Is there a way to manipulate the way in which the colors on the vertices are interpolated between one another?

![grafik](https://user-images.githubusercontent.com/13386367/74241562-94baaf00-4cdc-11ea-8a34-8f26e6b1914a.png)


Here, I get these quite visible lines between the nodes.
One thing I could do is to refine the mesh itself and apply more known colors to the vertices but that would be really time consuming (each point has to be calculated).

**Comment by holance** [comment]
The only way is using texture sampling instead of using vertex color. Vertex color is not going to be accurate, since its interpolation is just simple linear.

**Comment by leonbohmann** [comment]
Having a look at this a couple of years later I am (again) trying to figure this out! Is there a new feature or gradient I can use?

**Comment by holance** [comment]
You need to do texture sampling.

**Comment by leonbohmann** [comment]
How would I do that?

**Comment by holance** [comment]
Assign texture coordinates for each vertices and assign a gradient texture map as diffuse map in material.
